### PR TITLE
docs(validator): validate withdrawal storage root against state root

### PR DESCRIPTION
**Pull request by seolaoh**
# Description
 
Update the specs document that is not reflecting previous code changes.

Related #76

### CODE REVIEW

File: specs/challenge.md [hunk 1]
```diff
@@ -302,11 +297,10 @@ The `_pair[4]` and `_pair[5]` contain the public input, which must be processed
 5. Verify that the `nextBlockHash` in `_srcOutputRootProof` matches the block hash derived from `_publicInput` and
    `_rlps`.
 6. Verify that the `withdrawalStorageRoot` in `_dstOutputRootProof` is contained in `stateRoot` in
-  `_dstOutputRootProof`.
+  `_dstOutputRootProof` using `merkleProof`.
 7. If the length of transaction hashes in `_publicInput` is less than `MAX_TXS`, fill it with `DUMMY_HASH`.
-8. Verify the computation of the `publicInputHash` by comparing it with the `expectedPublicInputHash`.
-   The `publicInputHash` is derived from the `_publicInput` mentioned earlier, while the `expectedPublicInputHash`
-   is constructed using `_pair[4]` and `_pair[5]`.
+8. Verify the computation of the `publicInputHash` by comparing it with `_zkproof` and `_pair`.
```
[the remark this answers is not in the document]
Comment (seolaoh) [reply specs/challenge.md:302]: How about like below?
```
8. Verify the `_zkproof` using `_pair` and `publicInputHash`. The `publicInputHash` is derived from the `_publicInput`
   mentioned earlier, while `_zkproof` and `_pair` are submitted by the challenger.
```

[the remark this answers is not in the document]
Comment (0xHansLee) [reply specs/challenge.md:302]: Good. It's better than I wrote.

File: specs/challenge.md [hunk 2]
```diff
@@ -302,11 +297,10 @@ The `_pair[4]` and `_pair[5]` contain the public input, which must be processed
...
+   The `publicInputHash` is derived from the `_publicInput` mentioned earlier.
```
[the remark this answers is not in the document]
Comment (seolaoh) [reply specs/challenge.md]: I applied it also. Please check again :)

**Review by 0xHansLee — APPROVED** [review]
LGTM

**Review by Pangssu — APPROVED** [review]
LGTM

**Pull request merged**